src/libraries/System.Memory/tests/Span/EnumerateRunePositions.cs

```diff
@@ -0,0 +1,248 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class SpanTests
+    {
+        private static void RunePosition_TestProps(Rune rune, int startIndex, int length, bool wasReplaced, RunePosition runePosition)
+        {
+            Assert.Equal(rune, runePosition.Rune);
+            Assert.Equal(startIndex, runePosition.StartIndex);
+            Assert.Equal(length, runePosition.Length);
+            Assert.Equal(wasReplaced, runePosition.WasReplaced);
+        }
+
+        private static void RunePosition_TestEquals(RunePosition expected, RunePosition runePosition)
+        {
+            if (expected.Rune == runePosition.Rune && expected.StartIndex == runePosition.StartIndex &&
+                expected.Length == runePosition.Length && expected.WasReplaced == runePosition.WasReplaced)
+            {
+                Assert.Equal(expected, runePosition);
+                Assert.Equal(runePosition, expected);
+
+                Assert.True(expected.Equals(runePosition));
+                Assert.True(runePosition.Equals(expected));
+
+                Assert.True(((object)expected).Equals(runePosition));
+                Assert.True(((object)runePosition).Equals(expected));
+
+                Assert.True(expected == runePosition);
+                Assert.True(runePosition == expected);
+
+                Assert.False(expected != runePosition);
+                Assert.False(runePosition != expected);
+
+                Assert.Equal(expected.GetHashCode(), runePosition.GetHashCode());
+            }
+            else
+            {
+                Assert.NotEqual(expected, runePosition);
+                Assert.NotEqual(runePosition, expected);
+
+                Assert.False(expected.Equals(runePosition));
+                Assert.False(runePosition.Equals(expected));
+
+                Assert.False(((object)expected).Equals(runePosition));
+                Assert.False(((object)runePosition).Equals(expected));
+
+                Assert.False(expected == runePosition);
+                Assert.False(runePosition == expected);
+
+                Assert.True(expected != runePosition);
+                Assert.True(runePosition != expected);
+
+                Assert.NotEqual(expected.GetHashCode(), runePosition.GetHashCode());
```

Comment:
This assert should be removed. Two objects which are not equal can still have the same hash code.